### ISSUE TITLE
Network very unstable

### ISSUE THREAD

**Issue by mulho** [abiosoft/colima#280]
Hi
I just got started with Colima as a replacement for Docker Desktop on an Apple M1.
I'm experimenting with colima via Skaffold. It does seem to work, but every build fails with some network timeouts, and I have to rerun it multiple times so it can complete.
Some retries will work, and some will fail. And my network seems to be pretty decent (8ms, 530/400Mbps on Okla).
I tried this on cable and on Wifi.

This is what it looks like:
```
The push refers to repository [myacr.azurecr.io/my-service]
5f70bf18a086: Preparing
123eebb2977c: Preparing
c35e75c0d158: Preparing
98e743964c99: Preparing
98090211029b: Preparing
4fc242d58285: Preparing
4fc242d58285: Waiting
c35e75c0d158: Retrying in 5 seconds
5f70bf18a086: Layer already exists
98e743964c99: Layer already exists
98090211029b: Layer already exists
c35e75c0d158: Retrying in 4 seconds
c35e75c0d158: Retrying in 3 seconds
c35e75c0d158: Retrying in 2 seconds
c35e75c0d158: Retrying in 1 second
4fc242d58285: Layer already exists
123eebb2977c: Pushed
c35e75c0d158: Retrying in 10 seconds
c35e75c0d158: Retrying in 9 seconds
c35e75c0d158: Retrying in 8 seconds
c35e75c0d158: Retrying in 7 seconds
c35e75c0d158: Retrying in 6 seconds
c35e75c0d158: Retrying in 5 seconds
c35e75c0d158: Retrying in 4 seconds
c35e75c0d158: Retrying in 3 seconds
c35e75c0d158: Retrying in 2 seconds
c35e75c0d158: Retrying in 1 second
c35e75c0d158: Retrying in 15 seconds
c35e75c0d158: Retrying in 14 seconds
c35e75c0d158: Retrying in 13 seconds
c35e75c0d158: Retrying in 12 seconds
c35e75c0d158: Retrying in 11 seconds
c35e75c0d158: Retrying in 10 seconds
c35e75c0d158: Retrying in 9 seconds
c35e75c0d158: Retrying in 8 seconds
c35e75c0d158: Retrying in 7 seconds
c35e75c0d158: Retrying in 6 seconds
c35e75c0d158: Retrying in 5 seconds
c35e75c0d158: Retrying in 4 seconds
c35e75c0d158: Retrying in 3 seconds
c35e75c0d158: Retrying in 2 seconds
c35e75c0d158: Retrying in 1 second
c35e75c0d158: Retrying in 20 seconds
c35e75c0d158: Retrying in 19 seconds
c35e75c0d158: Retrying in 18 seconds
c35e75c0d158: Retrying in 17 seconds
c35e75c0d158: Retrying in 16 seconds
c35e75c0d158: Retrying in 15 seconds
c35e75c0d158: Retrying in 14 seconds
c35e75c0d158: Retrying in 13 seconds
c35e75c0d158: Retrying in 12 seconds
c35e75c0d158: Retrying in 11 seconds
c35e75c0d158: Retrying in 10 seconds
c35e75c0d158: Retrying in 9 seconds
c35e75c0d158: Retrying in 8 seconds
c35e75c0d158: Retrying in 7 seconds
c35e75c0d158: Retrying in 6 seconds
c35e75c0d158: Retrying in 5 seconds
c35e75c0d158: Retrying in 4 seconds
c35e75c0d158: Retrying in 3 seconds
c35e75c0d158: Retrying in 2 seconds
c35e75c0d158: Retrying in 1 second
build [my-service] failed: could not push image "myacr.azurecr.io/my-service:release-xxxx-dirty": dial tcp: lookup myacr.azurecr.io on 192.168.5.3:53: read udp 192.168.5.15:38056->192.168.5.3:53: i/o timeout
```
Retrying works, but it's annoying to have to retry multiple times on every build.

Note: To try to make sure this is not an issue with my router, I took it out of the game, and plugged my computer directly into my modem. Still same inconsistencies.

**Comment by mulho** [comment]
Oh, and eventually I also get this:
```
failed to solve with frontend dockerfile.v0: failed to create LLB definition: failed to do request: Head "https://registry-1.docker.io/v2/library/alpine/manifests/3.15": dial tcp: lookup registry-1.docker.io on 192.168.5.3:53: read udp 192.168.5.15:43813->192.168.5.3:53: i/o timeout
```

**Comment by rfay** [comment]
I have seen behavior like that and done a `colima stop && colima start` to resolve; also moved to colima HEAD (but you have to do a new .lima/.colima when you move)

**Comment by mulho** [comment]
Thanks @rfay 
Stop/start didn't help at all, but `HEAD` fixed it!
Thanks a lot

As a side note, when I upgraded to HEAD I deleted both `~/.colima` and `~/.lima`, and it started failing with
```
Cannot connect to the Docker daemon at unix:///Users/me/.colima/docker.sock. Is the docker daemon running?
```
So I noticed the file is now under a different location, and I had to run
```
DOCKER_HOST=unix:///Users/me/.colima/default/docker.sock
```

**Comment by abiosoft** [comment]
Can you upgrade to https://github.com/abiosoft/colima/releases/tag/v0.4.0 and see if the behaviour is also better.

Thanks.

**Comment by mortensassi** [comment]
`colima start --dns=1.1.1.1` helped me alot on my m1 macbook

**Comment by abiosoft** [comment]
This should be resolved in v0.4.x, will re-open if that is not the case.

**Comment by alvinpascual-symend** [comment]
I have this exact issue when I'm pulling many container images from GitHub Packages and nuget (.NET). I'm running colima 0.4.2.

![image](https://user-images.githubusercontent.com/101598335/171522800-54cd6a15-5c8a-42ab-ba49-dfbb6dfc189f.png)
